modules/core/src/main/java/org/apache/ignite/internal/processors/platform/client/cache/ClientCacheKeyRequest.java

```diff
@@ -47,6 +50,28 @@ public abstract class ClientCacheKeyRequest extends ClientCacheDataRequest imple
 
     /** {@inheritDoc} */
     @Override public final ClientResponse process(ClientConnectionContext ctx) {
+        updateMetrics(ctx);
+
+        // Process request in overriden method.
+        return process0(ctx);
+    }
+
+    /** {@inheritDoc} */
+    @Override public final IgniteInternalFuture<ClientResponse> processAsync(ClientConnectionContext ctx) {
+        updateMetrics(ctx);
+
+        // Process request in overriden method.
+        return processAsync0(ctx);
+    }
+
+    /** {@inheritDoc} */
+    @Override public boolean isAsync(ClientConnectionContext ctx) {
+        // Every cache data request on the transactional cache can lock the thread, even with implicit transaction.
+        return cacheDescriptor(ctx).cacheConfiguration().getAtomicityMode() == CacheAtomicityMode.TRANSACTIONAL;
```

Comment:
I think async processing can improve scalability even for ATOMIC caches, do you agree?

Reply:
Unfortunately, implicit operations also can block the thread. So, ignore my last proposal.